pkg/networking/cilium/templater_test.go

```diff
@@ -385,3 +385,20 @@ func TestGenerateNetworkPolicyManifestInvalidKubeVersion(t *testing.T) {
 	tt.Expect(err).To(HaveOccurred(), "templater.GenerateUpgradeManifest() should fail")
 	tt.Expect(err).To(MatchError(ContainSubstring("invalid major version in semver")))
 }
+
+func TestTemplaterGenerateManifestForSingleNodeCluster(t *testing.T) {
+	tt := newtemplaterTest(t)
+	tt.spec.Cluster.Spec.WorkerNodeGroupConfigurations = nil
+	tt.spec.Cluster.Spec.ControlPlaneConfiguration.Count = 1
+
+	tt.h.EXPECT().
+		Template(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+		Do(func(_ interface{}, _ interface{}, _ interface{}, _ interface{}, values map[string]interface{}, _ interface{}) {
+			tt.Expect(reflect.ValueOf(values["operator"]).MapIndex(reflect.ValueOf("replicas")).Interface().(int)).To(Equal(1))
+		})
+
+	_, err := tt.t.GenerateManifest(tt.ctx, tt.spec)
+	if err != nil {
+		t.Fatalf("failed to generate manifest: %v", err)
+	}
```

Comment:
do you want to add an assertion for the contents of the manifest?

Reply:
Good idea!